diff --git a/src/lj_jit.h b/src/lj_jit.h
index af50e5c1..f971ec0f 100644
--- a/src/lj_jit.h
+++ b/src/lj_jit.h
@@ -104,14 +104,20 @@
 
 /* -- JIT engine parameters ----------------------------------------------- */
 
+#if LJ_TARGET_WINDOWS || LJ_64
+#define JIT_P_sizemcode_DEFAULT		64
+#else
+#define JIT_P_sizemcode_DEFAULT		32
+#endif
+
 /* Optimization parameters and their defaults. Length is a char in octal! */
 #define JIT_PARAMDEF(_) \
-  _(\010, maxtrace,	1000)	/* Max. # of traces in cache. */ \
-  _(\011, maxrecord,	4000)	/* Max. # of recorded IR instructions. */ \
+  _(\010, maxtrace,	8000)	/* Max. # of traces in cache. */ \
+  _(\011, maxrecord,	16000)	/* Max. # of recorded IR instructions. */ \
   _(\012, maxirconst,	500)	/* Max. # of IR constants of a trace. */ \
   _(\007, maxside,	100)	/* Max. # of side traces of a root trace. */ \
   _(\007, maxsnap,	500)	/* Max. # of snapshots for a trace. */ \
-  _(\011, minstitch,	0)	/* Min. # of IR ins for a stitched trace. */ \
+  _(\011, minstitch,	3)	/* Min. # of IR ins for a stitched trace. */ \
   \
   _(\007, hotloop,	56)	/* # of iter. to detect a hot loop/call. */ \
   _(\007, hotexit,	10)	/* # of taken exits to start a side trace. */ \
@@ -123,9 +129,9 @@
   _(\011, recunroll,	2)	/* Min. unroll for true recursion. */ \
   \
   /* Size of each machine code area (in KBytes). */ \
-  _(\011, sizemcode,	64) \
+  _(\011, sizemcode,	JIT_P_sizemcode_DEFAULT) \
   /* Max. total size of all machine code areas (in KBytes). */ \
-  _(\010, maxmcode,	2048) \
+  _(\010, maxmcode,	40960) \
   /* End of list. */
 
 enum {
